### ISSUE TITLE
⌨ Notifications long list load more

### ISSUE THREAD

**Issue by vd1992** [GCTC-NTGC/gc-digital-talent#10640]
## ⌨ Accessibility Issue

Tabbing through a long list on the view all notifications page, reach "load more" and notice activating it shifts focus below the notifications section rather than onto the newly loaded notifications 

The focus disappears for a bit until you hit tab again, then it lands on "Contact us", guessing it persists on the now invisible element

### WCAG 2.1 Criteria

List any relevant criteria that is failing.

## 🦋 Expected Behaviour

Focus lands on new notifications or at least still in the list maybe a prior item

## 📋 Steps to Reproduce

1. Go to `/en/applicant/notifications` and have a lot of notifications
2. Tab through and hit "load more"

## 📸 Screenshot

Before

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/d61b9ce3-533c-48fc-a098-eb682d9f180a)

After

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/b91bd24e-e473-46c4-8010-6715dd15beef)

## 🙋‍♀️ Proposed Solution

(optional) Do you have a proposed solution?

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the issue was addressed.

- [ ] Focus should automatically move to the first newly-added notification every time the "load more" button is used
